future-apps/payroll/contracts/Payroll.sol

```diff
@@ -687,7 +687,7 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
      * @param _type Payment type to be performed
      * @return True if there was at least one token transfer
      */
-    function _transferTokensAmount(uint256 _employeeId, PaymentType _type, uint256 _totalAmount) internal returns (bool somethingPaid) {
+    function _transferTokensAmount(uint256 _employeeId, PaymentType _type, uint256 _totalAmount) internal nonReentrant returns (bool somethingPaid) {
```

Comment:
We should protect `determineAllocation()` as well, since that's the one that can change the later payments on an re-entrancy. It may be safest just to attach `nonReentrant` on all methods with the `employeeExists` modifier to prevent things like the address changing, etc.